pkg/controllers/plugin/util.go

```diff
@@ -4,16 +4,91 @@
 package plugin
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 
 	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	greenhouseapis "github.com/cloudoperators/greenhouse/pkg/apis"
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+	"github.com/cloudoperators/greenhouse/pkg/clientutil"
 )
 
+// initClientGetter returns a RestClientGetter for the given Plugin.
+// If the Plugin has a clusterName set, the RestClientGetter is initialized from the cluster secret.
+// Otherwise, the RestClientGetter is initialized with in-cluster config
+func initClientGetter(
```

Comment:
I think this needs to be properly refactored. The `initClientGetter()` ingesting a `PluginStatus` was a very proper use case of the `helm_controller`.

See suggestions below...

Reply:
The callings of `initClientGetter()` in 
https://github.com/cloudoperators/greenhouse/blob/main/pkg/controllers/plugin/helm_controller.go#L120
and in plugin_workload_status_controller.go need to be adjusted accordingly

Reply:
Tracked here and planned for next sprint. Should be a small thing and we can straighten this out for all 3 controller requiring the function: https://github.com/cloudoperators/greenhouse/issues/488